src/test/scala/io/citrine/lolo/bags/BaggerTest.scala

```diff
@@ -100,6 +102,39 @@ class BaggerTest {
     )
   }
 
+  /**
+    * Test that the bagged learner can be interrupted
+    */
+  @Test
+  def testInterrupt(): Unit = {
+    val trainingData = TestUtils.generateTrainingData(2048, 12, noise = 0.1, function = Friedman.friedmanSilverman)
+    val DTLearner = new RegressionTreeLearner(numFeatures = 3)
+    val baggedLearner = new Bagger(DTLearner, numBags = trainingData.size)
+
+    val tmpPool = Executors.newFixedThreadPool(1)
+    val fut: Future[BaggedTrainingResult] = tmpPool.submit(
+      new Callable[BaggedTrainingResult] {
+        override def call() = {
+          println("Starting to run the future")
+          val res = baggedLearner.train(trainingData)
+          assert(false, "Training was not terminated")
+          res
+        }
+      }
+    )
+    Thread.sleep(1000)
+    assert(fut.cancel(true), "Failed to cancel future")
```

Comment:
Can you add a timer to ensure that the time it takes from line 126 to 136 is <2secs or so?